dso-l2/src/main/java/com/tc/l2/state/StateManagerImpl.java

```diff
@@ -141,7 +136,13 @@ public void startElection() {
     } else if (state == START_STATE || state == PASSIVE_STANDBY) {
       runElection();
     } else {
-      info("Ignoring Election request since not in right state");
+      info("Ignoring Election request since not in right state: " + this.state);
+    }
+    // Now that we are ready, put us into the started state and notify anyone stuck waiting.
+    synchronized (this) {
+      Assert.assertFalse(this.didStartElection);
+      this.didStartElection = true;
```

Comment:
doesn't this need to go before the runElection() call?  If the thread stalls after runElection(), I think the reply could be missed.